src/helper/tools/fill-tool.js

```diff
@@ -43,21 +45,30 @@ class FillTool extends paper.Tool {
                 item.lastSegment.point.getDistance(item.firstSegment.point) < 8;
         };
         return {
-            segments: true,
+            segments: false,
             stroke: true,
-            curves: true,
+            curves: false,
             fill: true,
             guide: false,
             match: function (hitResult) {
+                // Allow fills to be hit only if the item has a fill already or the path is closed/nearly closed
+                const hitFill = hitResult.item.hasFill() || hitResult.item.closed || isAlmostClosedPath(hitResult.item);
                 if (hitResult.item instanceof paper.Path &&
-                    (hitResult.item.hasFill() || hitResult.item.closed || isAlmostClosedPath(hitResult.item))) {
+                    // Disallow hits that don't qualify for the fill criteria, but only if they're fills
+                    (hitFill || hitResult.type !== 'fill')) {
                     return true;
                 }
                 if (hitResult.item instanceof paper.PointText) {
                     return true;
                 }
             },
             hitUnfilledPaths: true,
+            // If the color is transparent/none, then we need to be able to hit "invisible" outlines so that we don't
+            // prevent ourselves from hitting an outline when we make it transparent via the fill preview, causing it to
+            // flicker back and forth between transparent/its previous color as we hit it, then stop hitting it, etc.
+            // If the color *is* visible, then don't hit "invisible" outlines, since this would add visible outlines to
+            // non-outlined shapes when you hovered over where their outlines would be.
```

Comment:
In Scratch 2.0, you can use the fill tool to add strokes to paths that don't have strokes ("invisible strokes"), and the stroke width would be set to some arbitrary relatively thick width. We can reproduce this by hitting curves and not strokes when the stroke is invisible, and overwriting whatever stroke width was on the invisible stroke before. If we want to change the behavior from 2.0, let's remember to ask about it in the paint huddle

Reply:
We conferred with design and decided not to "fill" invisible outlines for now